zipkin-server/src/main/java/zipkin/server/internal/ZipkinHttpCollector.java

```diff
@@ -140,7 +148,13 @@
 
   static void error(HttpServerExchange exchange, Throwable e) {
     String message = e.getMessage();
-    int code = message == null || message.startsWith("Cannot store") ? 500 : 400;
+    int code;
+    if (e instanceof FilterActivatedException) {
+      FilterActivatedException filterException = (FilterActivatedException)e;
+      code = filterException.getHttpCode();
+    } else {
+      code = message == null || message.startsWith("Cannot store") ? 500 : 400;
```

Comment:
Is "Cannot store" an important string? Should it be a constant?

Reply:
It is an important String, but I rather not touch it because it was already there. I would recommend that it be changed to typed exception rather than relying on a String. I'll note to bring this up with Adrian and would prefer to make this change in a separate PR.